### DISCUSSION TITLE
@PreAuthorize 를 api 에 적극 사용하는 거 어떨까요

### DISCUSSION THREAD

**Discussion by kang20** [LikeLion12th-OMEB/Backend#41] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/9f079494-8638-4582-bbeb-401b1ffde8c0)

위 사진처럼 @PreAuthorize 를 사용하면 api 메서드를 호출하기 전에 () 안에 정규식에 맞는 권한 검사를 합니다.
security Config 설정을 통해 filter에서도 검사할 수 있긴하지만 현재 모든 권한을 허용하기도 했고 api controller 를 볼때 명시적으로 권한 검사를 할 수 있을거 같아서 적용하면 좋을 거 같습니다 

- 표현식 정의입니다

![image](https://github.com/user-attachments/assets/9dd89ff8-6e20-4274-b875-4bbb39e86e5d)

해당 권한 검사가 실패하면 AccessDeniedException 에러를 반환하고 해당 에러코드는 

 `    ACCESS_DENIED(HttpStatus.UNAUTHORIZED, "AUTH_006", "인증되지 않은 사용자입니다."), `

를 사용할 예정입니다

의견 어떠신가요

**Comment by siwan9** [comment, marked as answer]
괜찮은 것 같습니다.
에러코드는 기존에 있는 AUTH_0004에 HttpStatus.UNAUTHORIZED로 변경하여 사용해도 될 것 같네요
